### PR TITLE
Templated BVertex plugins

### PR DESCRIPTION
**Pull request by ferencek**
This PR templates the BVertexFilter and BtoCharmDecayVertexMerger plugins in order to make them usable in the candidate-based framework

No impact on the standard reconstruction is expected.

@imarches @cvernier

### CODE REVIEW

File: RecoBTag/SecondaryVertex/plugins/BtoCharmDecayVertexMerger.cc [hunk 1]
```diff
@@ -20,55 +22,78 @@
...
+template<typename VTX>
+BtoCharmDecayVertexMergerT<VTX>::BtoCharmDecayVertexMergerT(const edm::ParameterSet &params) :
        token_primaryVertex(consumes<reco::VertexCollection>(params.getParameter<edm::InputTag>("primaryVertices"))),
-       token_secondaryVertex(consumes<reco::VertexCollection>(params.getParameter<edm::InputTag>("secondaryVertices"))),
-       minDRForUnique(params.getUntrackedParameter<double>("minDRUnique",0.4)),
-       vecSumIMCUTForUnique(params.getUntrackedParameter<double>("minvecSumIMifsmallDRUnique",5.5)),
-       minCosPAtomerge(params.getUntrackedParameter<double>("minCosPAtomerge",0.99)),
-       maxPtreltomerge(params.getUntrackedParameter<double>("maxPtreltomerge",7777.0))
+       token_secondaryVertex(consumes<edm::View<VTX> >(params.getParameter<edm::InputTag>("secondaryVertices"))),
+       maxDRForUnique(params.getParameter<double>("maxDRUnique")),
```
Comment (slava77) [inline RecoBTag/SecondaryVertex/plugins/BtoCharmDecayVertexMerger.cc]: this is good
untracked -> tracked

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/11381	d61c1f2
- code changes are rather technical and no changes are expected in the currently running setup(s)
- jenkins tests complete and comparisons with baseline show no differences

**Comment by davidlange6** [comment]
+1

**Pull request merged**